File: package/main/src/Validate/isPerfectSquare.ts
/**
 * Determines if a given integer is a perfect square
 * @param {number} number_ - Integer to check
 * @returns {boolean} true if the number is a perfect square, false otherwise
 * @example isPerfectSquare(16); // true
 * isPerfectSquare(25); // true
 * isPerfectSquare(10); // false
 * isPerfectSquare(-4); // false
 * isPerfectSquare(2.25); // false (non-integer input)
 */
export const isPerfectSquare = (number_: number): boolean => {
  if (number_ < 0) {
    return false;
  }

  const sqrt = Math.sqrt(number_);
  return sqrt === Math.floor(sqrt);
};

File: package/umt_python/src/math/is_perfect_square.py
import math


def is_perfect_square(n: int) -> bool:
    """
    Determines if a given integer is a perfect square.

    Args:
        n: Integer to check.

    Returns:
        True if the number is a perfect square, False otherwise.

    Example:
        >>> is_perfect_square(16)
        True
        >>> is_perfect_square(25)
        True
        >>> is_perfect_square(10)
        False
        >>> is_perfect_square(-4)
        False
    """
    if n < 0:
        return False

    sqrt = math.sqrt(n)
    return sqrt == math.floor(sqrt)
